backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAO.java

```diff
@@ -198,7 +199,9 @@ default DatasetVersion getDatasetVersionFromRequest(
       throws ModelDBException {
     DatasetVersion.Builder datasetVersionBuilder =
         DatasetVersion.newBuilder()
-            .setId(UUID.randomUUID().toString())
+            .setId(
+                VersioningUtils.createRepoCommitCompositeIdString(
+                    Long.parseLong(request.getDatasetId()), UUID.randomUUID().toString()))
```

Comment:
not sure about the random UUID here, should n't this be

Reply:
Fixed